lang/cs/Org.Apache.REEF.Common/Telemetry/DriverMetrics.cs

```diff
@@ -26,14 +26,17 @@ namespace Org.Apache.REEF.Common.Telemetry
     /// </summary>
     public sealed class DriverMetrics : IDriverMetrics
```

Comment:
Since you've added a generic "IMetrics" why not make the DriverMetrics an IMetrics as well. This should be done the same way you did Counters.